### PR TITLE
[build] Use `rpath` for BAT and ROOT dependencies

### LINKED ISSUE #186: on MacOS BAT gives "rpath" err after upgrading to Sierra (10.12)
Hi, 

we are trying to install and run examples of BAT on Mac Sierra (10.12). and it's giving 

picture_error

error.

we checked that on Mac EI Capitan (10.11) it's working well.

we did some debug with `otool -L` on examples/basic/binomial package. here is the result:

on Sierra (where BAT does not run)
otool -L .libs/runEfficiency gives
otool_sierra

otool -l .libs/runEfficiency  gives
[otool_ml_EICapitan.zip](https://github.com/bat/bat/files/522233/otool_ml_EICapitan.zip)

on EI Capitan (where BAT works where) 
otool -L .libs/runEfficiency gives
otool_eicapitan
otool -l .libs/runEfficiency  gives

[otool_ml.zip](https://github.com/bat/bat/files/522232/otool_ml.zip)

it's strange that on Sierra BAT decided to use rpath, while LC_RPATH is not written in the executables.

at the same time, we found that it's possible to execute the .libs/runEfficiency directly without the wrapper, except some error shown. here is the output.
[Archive.zip](https://github.com/bat/bat/files/522252/Archive.zip)

could you help us identify the reason? btw do you have plans to port BAT to home-brew?

### PR DESCRIPTION
**Pull request by fredRos**
* update `INSTALL.md`: setting `LD_LIBRARY_PATH` is not mandatory
  anymore
* fixes #186
* `rpath` avoid accidentally loading a different version of root than
  that for which BAT was configured

### CODE REVIEW

**Comment by fredRos** [comment]
I ran `make check` and it worked fine on my machine, before merging, I'd ask @greenwald to do the same on his mac os sierra

**Comment by oschulz** [comment]
I'm not sure if we should do this: For once, it's a bit non-standard - most software I know is installed without rpath and requires LD_LIBRARY_PATH to be set, at least on Linux. Also, I worry that we'll make things less transparent to the user - suddenly, BAT works even though the wrong ROOT version is loaded. When looking for the source of some linker trouble (let's say in a project that used ROOT, BAT, and other libs), this may confuse less experienced people even more.

**Comment by fredRos** [comment]
> Also, I worry that we'll make things less transparent to the user - suddenly, BAT works even though the wrong ROOT version is loaded. When looking for the source of some linker trouble (let's say in a project that used ROOT, BAT, and other libs), this may confuse less experienced people even more.

I disagree. Now things are clearer, and we can state this in `INSTALL.md`: at configure time, a root version is chosen. Whenever you use BAT, it uses that exact version of root unless `PRELOAD` is used, which is left to the experts.

What happened before and is avoided now is that you compile with say root 5 and then load root 6 on the `LD_LIBRARY_PATH`. This happened to me as a developer, and I usually found out quickly why things didn't work. Or think of the linker problem @greenwald just had after a system update. These things really confuse people, even developers, and we should avoid that. With `rpath`, we can.

The situation is different for a distribution, Debian banned `rpath` https://wiki.debian.org/RpathIssue but I think we don't fall into the diamond trap: we don't rely on libraries A.so.**1** and B.so.1 and A depends on B.so.**2**

So I agree this is not an easy decision but I'd prefer the rpath solution. If we decide to make packages for debian, we'd have to undo that.

**Comment by oschulz** [comment]
> What happened before and is avoided now is that you compile with say root 5 and then load root 6 on the LD_LIBRARY_PATH. This happened to me as a developer, and I usually found out quickly why things didn't work. Or think of the linker problem @greenwald just had after a system update.

I think we'd cause even more confusion if BAT works even though the wrong ROOT version is on PATH and LD_LIBRARY_PATH. What would happen if someone tries to run a ROOT script that uses BAT, for example? I guess things would crash horribly, but it would be even harder for the user to figure out why.

> The situation is different for a distribution, Debian banned rpath https://wiki.debian.org/RpathIssue

Sure, we're not a distribution, but I can't think of any software right now that forces rpath (at least not on Linux).

> So I agree this is not an easy decision but I'd prefer the rpath solution. If we decide to make packages for debian, we'd have to undo that.

I don't think we should do something that we'd have to undo to make a Debian package. That seems a too much like a hack to me.

File: INSTALL.md [hunk 1]
```diff
@@ -226,11 +226,16 @@ for csh-compatible shells. On Mac OS X you might also need to setup
...
+Updating `$CPATH` is required if you work with interactive ROOT macros
+that use BAT (both for ROOT 5 and 6).
+
+The minimal setup does not require setting `LD_LIBRARY_PATH` if
+libraries or executables are linked to BAT using `bat-config --libs`
+because the directory from which the BAT libraries are loaded at
+runtime is set via the `rpath` option.
 
-Updating `$CPATH` is required if you work with ROOT macros that use
-BAT (both for ROOT 5 and 6).
```
Comment (oschulz) [inline INSTALL.md]: Isn't `CPATH` still required when the ROOT script contains include statements?

**Comment by fredRos** [comment]
I cannot say, @oschulz is the root expert. I just remember adding this
statement a while ago to make the macro examples work. I haven't used
root macros w/o bat myself. Feel free to correct this if needed.

**Comment by greenwald** [comment]
If we don't set `rpath`, what is the solution for the current Mac OS?

**Comment by fredRos** [comment]
The only solution I see for the current mac os is setting `rpath`.
@oschulz just wants to continue to use `LD_LIBRARY_PATH` on linux

**Comment by greenwald** [comment]
`rpath` is still only a search path, no? If one changes an installation of ROOT, so long as he installs it into the same location, it should be fine. Additionally, we can state in the documentation that if one updates ROOT, one should accordingly update BAT.

**Pull request merged**